test/tools/router_check/router.cc

```diff
@@ -0,0 +1,203 @@
+             -> bool { return compareVirtualCluster(tool_config, expected); }},
+        {"virtual_host_name", [this](ToolConfig& tool_config, const std::string& expected)
+                                  -> bool { return compareVirtualHost(tool_config, expected); }},
+        {"path_rewrite", [this](ToolConfig& tool_config, const std::string& expected)
+                             -> bool { return compareRewritePath(tool_config, expected); }},
+        {"host_rewrite", [this](ToolConfig& tool_config, const std::string& expected)
+                             -> bool { return compareRewriteHost(tool_config, expected); }},
+        {"path_redirect", [this](ToolConfig& tool_config, const std::string& expected)
+                              -> bool { return compareRedirectPath(tool_config, expected); }},
+    };
+
+    for (std::pair<std::string, std::function<bool(ToolConfig&, std::string)>> test : checkers) {
+      if (validate->hasObject(test.first)) {
+        std::string expected = validate->getString(test.first);
+        if (tool_config.route_ == nullptr) {
+          compareResults("", expected, test.first);
+        } else {
+          if (!test.second(tool_config, validate->getString(test.first))) {
+            no_failures = false;
+          }
+        }
+      }
+    }
+
+    if (validate->hasObject("header_fields")) {
+      for (const Json::ObjectPtr& header_field : validate->getObjectArray("header_fields")) {
+        if (!compareHeaderField(tool_config, header_field->getString("field"),
+                                header_field->getString("value"))) {
+          no_failures = false;
+        }
+      }
+    }
+  }
+  return no_failures;
+}
+
+void ToolConfig::parseFromJson(const Json::ObjectPtr& check_config) {
+  // Extract values from json object
+  Json::ObjectPtr input = check_config->getObject("input");
+
+  random_value_ = input->getInteger("random_value", 0);
+
+  // Add authority and path to header
+  headers_.addViaCopy(":authority", input->getString(":authority", ""));
+  headers_.addViaCopy(":path", input->getString(":path", ""));
+
+  // Add :method to header
+  headers_.addViaCopy(":method", input->getString(":method", "GET"));
+
+  // Add ssl header
+  headers_.addViaCopy("x-forwarded-proto", input->getBoolean("ssl", false) ? "https" : "http");
+
+  // Add internal route status to header
+  if (input->getBoolean("internal", false)) {
+    headers_.addViaCopy("x-envoy-internal", "true");
+  }
+
+  // Add additional headers
+  if (input->hasObject("additional_headers")) {
+    for (const Json::ObjectPtr& header_config : input->getObjectArray("additional_headers")) {
+      headers_.addViaCopy(header_config->getString("field"), header_config->getString("value"));
+    }
+  }
+}
+
+bool RouterCheckTool::compareCluster(ToolConfig& tool_config, const std::string& expected) {
+  std::string actual = "";
+  if (tool_config.route_->routeEntry() != nullptr) {
```

Comment:
nit: keep new line spacing consistent across Compare Methods. either had a new line here or remove the new lines in the other compare methods